packages/dapp/hooks/useGallery.js

```diff
@@ -41,39 +38,44 @@ const useGallery = () => {
         toBlock: 'latest',
       };
 
-      const logs = await provider.getLogs(filter);
+      const [batchSize, totalSupply] = await Promise.all([
+        Ethernauts.batchSize().then(Number),
+        Ethernauts.totalSupply().then(Number),
+      ]);
```

Comment:
Here, instead of asking the contract for each token, we just get the amount of tokens are minted and the batch size, and then with that we can calculate if any token is revealed by its ID.